### PR TITLE
task4602 notifications

### PR DESCRIPTION
**Pull request by laam55**
https://edu-redmine.framgia.vn/issues/4602

### CODE REVIEW

File: app/Http/Middleware/MarkNotificationAsRead.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+class MarkNotificationAsRead
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
```
Comment (framgiaciv3) [inline app/Http/Middleware/MarkNotificationAsRead.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Http/Middleware/MarkNotificationAsRead.php</br>**Line**: 18</br>Expected 1 space(s) after IF keyword; 0 found

File: app/Http/Middleware/MarkNotificationAsRead.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        if($request->has('read')) {
+            $notification = $request->user()->notifications()->where('id', $request->read)->first();
```
Comment (framgiaciv3) [inline app/Http/Middleware/MarkNotificationAsRead.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Http/Middleware/MarkNotificationAsRead.php</br>**Line**: 21</br>Expected 1 space(s) after IF keyword; 0 found

**Comment by laam55** [comment]
ready

File: resources/views/home/layout/header.blade.php [hunk 3]
```diff
@@ -71,6 +79,78 @@
                 </a>
             </li>
             @endif
+            <!-- inbox dialog -->
+            @if (Auth::check())
+                <div class="topbar-dialog inbox-dialog dno" id="inbox-dialog">
+                    <div class="header">
+                        <h3>inbox</h3>
```
Comment (BunnyPi04) [inline resources/views/home/layout/header.blade.php]: i18n

File: resources/views/home/master.blade.php [hunk 4]
```diff
@@ -12,6 +12,16 @@
 
     <link rel="stylesheet" href="{{ asset('css/app.css') }}">
     <link rel="stylesheet" href="{{ asset('css/template.css') }}">
+    <script>
+        window.Laravel = @php echo json_encode([
+            'csrfToken' => csrf_token(),
+        ]); @endphp
+    </script>
+    @if(!auth()->guest())
```
Comment (BunnyPi04) [inline resources/views/home/master.blade.php]: convention

File: resources/views/home/master.blade.php [hunk 5]
```diff
@@ -33,14 +43,46 @@
...
             $('.nuxt-progress').animate({ width: '100%' }, 1000, function() {
                 $('.nuxt-progress').css('opacity', 0);
             });
 
             $('body').addClass('users-page unified-theme');
             $('.alert-dismissible').fadeTo(3000, 500).slideUp(500);
+            $('.js-inbox-button').click(function(event) {
+                var mousePosY = ($(window).width() - ($('.js-inbox-button').offset().left + $('.js-inbox-button').outerWidth()));
+                $('.inbox-dialog').toggle(100).css({
+                    'right' : mousePosY - 40,
```
Comment (BunnyPi04) [inline resources/views/home/master.blade.php]: convention

File: resources/views/home/master.blade.php [hunk 6]
```diff
@@ -33,14 +43,46 @@
...
+                    'position' : 'absolute',
+                });
+            });
+
+            if(Laravel.userId) {
+                $.get('/user/notifications', function (data) {
+                    notifications = data.map(function(val) {
+                        val.data.id = val.id;
+                        return val.data;
```
Comment (BunnyPi04) [inline resources/views/home/master.blade.php]: thiếu dòng trắng trc return

File: resources/views/home/master.blade.php [hunk 7]
```diff
@@ -33,14 +43,46 @@
...
+                    });
+                });
+                window.Echo.private(`App.User.${Laravel.userId}`).notification((notification) => {
+                    console.log(notification);
```
Comment (BunnyPi04) [inline resources/views/home/master.blade.php]: xóa console.log đi nhé

**Comment by laam55** [comment]
ready

**Pull request merged**